franklin/core/helpers.py

```diff
@@ -68,3 +76,41 @@ def has_permission(self, request, view):
             is_valid_github = hmac.compare_digest(computed_secret, secret)
             return is_valid_github
         return False
+
+
+class SocialAuthentication(BaseAuthentication):
+    """
+    Used in DEFAULT_AUTHENTICATION_CLASSES settings for authentication.
+    This setting allows users to authenticate with only a Github oauth token.
+    """
+    def authenticate(self, request):
+        auth_header = get_authorization_header(request)\
+                            .decode(HTTP_HEADER_ENCODING)
+        auth = auth_header.split()
+        if not auth or auth[0].lower() != 'bearer':
+            return None
+        if len(auth) == 1:
+            msg = 'Credentials are malformed'
+            raise exceptions.AuthenticationFailed(msg)
+        oauth_token = auth[1]
+
+        social_user = UserSocialAuth.objects\
+                                    .filter(extra_data__contains=oauth_token)\
+                                    .first()
```

Comment:
This call could have a large performance hit if we have a lot of users in the DB. This will be done for every API call that is made.